### PR TITLE
DS-3898: ScaleVariableSet to catch duplicate factor level labels

### PR DESCRIPTION
**Pull request by chrisfacer**
ScaleVariableSet to catch duplicate factor level labels

### CODE REVIEW

File: tests/testthat/test-scalevariableset.R [hunk 1]
```diff
@@ -204,3 +204,44 @@ test_that("PickOneMulti with merge, hide, NET",
...
+    C = 3, B = 4, A = 5), c(A = 1, B = 2, C = 3, B = 4, A = 5
+    ), c(A = 1, B = 2, C = 3, B = 4, A = 5), c(A = 1, B = 2, 
+    C = 3, B = 4, A = 5), c(A = 1, B = 2, C = 3, B = 4, A = 5
+    ), c(A = 1, B = 2, C = 3, B = 4, A = 5)), codeframe = list(
+    A = 1, B = 2, C = 3, B = 4, A = 5, NET = c(1, 2, 3, 4, 5)), secondarycodeframe = list(
+    `Label 1` = 0L, `Label 2` = 1L, `Label 3` = 2L, `Label 4` = 3L, 
+    `Label 5` = 4L, `Label 6` = 5L, `Label 7` = 6L, `Label 8` = 7L, 
+    `Label 9` = 8L, `Label 10` = 9L, `Label 11` = 10L, `Label 12` = 11L), transposed = TRUE, class = "data.frame")
+
+    expect_error(ScaleVariableSet(test.case, type = 'standardize', within.case = TRUE))
```
Comment (jrwishart) [inline tests/testthat/test-scalevariableset.R:246]: Ideally the error message could be checked against the matching string 
`The variable(s) that you have selected have duplicate category labels. You 
should edit the category labels to remove duplicates before using this
feature`

As an aside, something not quite right with this test case. It is valid in the sense that it correctly throws the error for the duplicated labels. However when run on `master` it throws a different error. 
```
 Invalid variable set provided; the length of the “variablevalues” 
attribute does not match  the number of variables in the variable set.
```

**Review by jrwishart — APPROVED** [review]

File: tests/testthat/test-scalevariableset.R (hunk 1 above)
Comment (jrwishart) [reply tests/testthat/test-scalevariableset.R:246]: So overall looks good, but might be worth tidying the test a bit.

**Comment by Jes-I-Guess** [comment]
Trying to install flipTransformations as described in the readme file but I get the following error :
```
Error in parse(outFile) : 
  /tmp/RtmpCle5Ux/R.INSTALL431b63a1ab60/flipTransformations/R/scalevariableset.R:141:42: unexpected input
140:     vv <- attr(x, "variablevalues")
141:     duplicate.levels <- vapply(vv, FUN = \
                                              ^
ERROR: unable to collate and parse R files for package ‘flipTransformations’
```

**Comment by GDQF** [comment]
Hello, same issue as @Jes-I-Guess

I want to install package flipPlots in R. This website (https://rdrr.io/github/Displayr/flipPlots/) mentions I can install it using 

`remotes::install_github("Displayr/flipTransformations", dependencies = NA)`

which I did and get the below error


```
* installing *source* package ‘flipTransformations’ ...
** using staged installation
** R
Error in parse(outFile) : 
  /tmp/RtmpDzEz0M/R.INSTALL73c388a2f4a/flipTransformations/R/scalevariableset.R:141:42: unexpected input
140:     vv <- attr(x, "variablevalues")
141:     duplicate.levels <- vapply(vv, FUN = \
                                              ^
ERROR: unable to collate and parse R files for package ‘flipTransformations’
```


I also tried to install flipTransformations first


```
* installing *source* package ‘flipTransformations’ ...
** using staged installation
** R
Error in parse(outFile) : 
  /tmp/Rtmp8DrHrN/R.INSTALL6057344fc1d34/flipTransformations/R/scalevariableset.R:141:42: unexpected input
140:     vv <- attr(x, "variablevalues")
141:     duplicate.levels <- vapply(vv, FUN = \
                                              ^
ERROR: unable to collate and parse R files for package ‘flipTransformations’
```

**Comment by jrwishart** [comment]
> Trying to install flipTransformations as described in the readme file but I get the following error :
> 
> ```
> Error in parse(outFile) : 
>   /tmp/RtmpCle5Ux/R.INSTALL431b63a1ab60/flipTransformations/R/scalevariableset.R:141:42: unexpected input
> 140:     vv <- attr(x, "variablevalues")
> 141:     duplicate.levels <- vapply(vv, FUN = \
>                                               ^
> ERROR: unable to collate and parse R files for package ‘flipTransformations’
> ```

Fixed now in https://github.com/Displayr/flipTransformations/commit/3324c109a7eaad10421799abc23c01bcef59e9d7. The `\(x)` lambda anonymous function syntax was used without the requirement for a newer R by mistake. Replaced it with the standard `function(x)` syntax preserving compatibility with earlier R. It should now install fine.

**Comment by Jes-I-Guess** [comment]
> > Trying to install flipTransformations as described in the readme file but I get the following error :
> > ```
> > Error in parse(outFile) : 
> >   /tmp/RtmpCle5Ux/R.INSTALL431b63a1ab60/flipTransformations/R/scalevariableset.R:141:42: unexpected input
> > 140:     vv <- attr(x, "variablevalues")
> > 141:     duplicate.levels <- vapply(vv, FUN = \
> >                                               ^
> > ERROR: unable to collate and parse R files for package ‘flipTransformations’
> > ```
> 
> Fixed now in [3324c10](https://github.com/Displayr/flipTransformations/commit/3324c109a7eaad10421799abc23c01bcef59e9d7). The `\(x)` lambda anonymous function syntax was used without the requirement for a newer R by mistake. Replaced it with the standard `function(x)` syntax preserving compatibility with earlier R. It should now install fine.

Issue now solved. Thanks for looking into it, much appreciated !

**Pull request merged**